ADAL/src/cache/ADTokenCacheAccessor.m

```diff
@@ -342,52 +359,52 @@ - (void)removeItemFromCache:(ADTokenCacheItem *)cacheItem
-        if ([refreshToken isEqualToString:existing.refreshToken])//If still there, attempt to remove
-        {
-            AD_LOG_VERBOSE_F(@"Token cache store", correlationId, @"Tombstoning cache for resource: %@", cacheItem.resource);
-            //update tombstone property before update the tombstone in cache
-            [existing makeTombstone:@{ @"correlationId" : [correlationId UUIDString],
-                                       @"errorDetails" : [error errorDetails],
-                                       @"protocolCode" : [error protocolCode] }];
-            [_dataSource addOrUpdateItem:existing correlationId:correlationId error:nil];
-            removed = YES;
-        }
+        return;
     }
     
-    if (!removed)
+    NSUUID* correlationId = [context correlationId];
+    
+    ADTokenCacheItem* existing = [_dataSource getItemWithKey:cacheKey
+                                                      userId:cacheItem.userInformation.userId
+                                               correlationId:correlationId
+                                                       error:nil];
+    if (!existing)
     {
-        //Now try finding a broad refresh token in the cache and tombstone it accordingly
-        ADTokenCacheKey* broadKey = [ADTokenCacheKey keyWithAuthority:_authority
-                                                             resource:nil
-                                                             clientId:cacheItem.clientId
-                                                                error:nil];
-        if (broadKey)
-        {
-            ADTokenCacheItem* broadItem = [_dataSource getItemWithKey:broadKey userId:cacheItem.userInformation.userId correlationId:correlationId error:nil];
-            if (broadItem && [refreshToken isEqualToString:broadItem.refreshToken])//Remove if still there
-            {
-                AD_LOG_VERBOSE_F(@"Token cache store", correlationId, @"Tombstoning multi-resource refresh token for authority: %@", _authority);
-                //update tombstone property before update the tombstone in cache
-                [broadItem makeTombstone:@{ @"correlationId" : [correlationId UUIDString],
-                                            @"errorDetails" : [error errorDetails],
-                                            @"protocolCode" : [error protocolCode] }];
-                [_dataSource addOrUpdateItem:broadItem correlationId:correlationId error:nil];
-            }
-        }
+        existing = [_dataSource getItemWithKey:[cacheKey mrrtKey]
+                                        userId:cacheItem.userInformation.userId
+                                 correlationId:correlationId
+                                         error:nil];
     }
-    ADTelemetryCacheEvent* event = [[ADTelemetryCacheEvent alloc] initWithName:AD_TELEMETRY_EVENT_TOKEN_CACHE_DELETE
-                                                                       context:context];
     
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the useless trailing whitespaces at the end of this line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=objc%3AS1131)